### PR TITLE
osd, mon: add pglog dups length

### PR DESCRIPTION
**Pull request by NitzanMordhai**
Add a new column LOG_DUPS, to the pg dump stats and log_dups_size to pg query
that will show the size of dups entries in pg log.

Fixes: https://tracker.ceph.com/issues/55836
Signed-off-by: Nitzan Mordechai <nmordech@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/osd/osd_types.cc [hunk 1]
```diff
@@ -3125,6 +3127,9 @@ void pg_stat_t::decode(ceph::buffer::list::const_iterator &bl)
       decode(objects_trimmed, bl);
       decode(snaptrim_duration, bl);
     }
+    if (struct_v >= 29) {
+      decode(log_dups_size, bl);
+    }
```
Comment (rzarzynski) [inline src/osd/osd_types.cc:3132]: What otherwise? It doesn't seem we initialize `log_dups_size` anyhow, even in ctor.

**Review by rzarzynski — COMMENTED** [review]
Generally LGTM. One question around the backward compatibility.

File: src/osd/osd_types.cc (hunk 1 above)
Comment (NitzanMordhai) [reply src/osd/osd_types.cc:3132]: Right, added it to the ctor, it will be zero in case of decode is less then 29

**Review by rzarzynski — APPROVED** [review]
LGTM.

File: src/osd/PeeringState.cc [hunk 2]
```diff
@@ -4164,6 +4165,8 @@ void PeeringState::append_log(
 
   psdout(10) << __func__ << " approx pg log length =  "
 	     << pg_log.get_log().approx_size() << dendl;
+  psdout(10) << __func__ << " dups pg log length =  "
+	     << pg_log.get_log().dups.size() << dendl;
```
Comment (rzarzynski) [inline src/osd/PeeringState.cc:4169]: How about dissecting this log entry into a tiny, dedicated commit? We could easily and safely backport it everywhere.

Comment (neha-ojha) [reply src/osd/PeeringState.cc:4169]: +1!

**Comment by sseshasa** [comment]
**Teuthology Test Result**
https://pulpito.ceph.com/yuriw-2022-06-27_15:15:17-rados-wip-yuri2-testing-2022-06-24-1331-distro-default-smithi/

**Related Failure:**
@NitzanMordhai The following failed job appears to be related to this PR. Can you please take a look?
**JobID:** https://pulpito.ceph.com/yuriw-2022-06-27_15:15:17-rados-wip-yuri2-testing-2022-06-24-1331-distro-default-smithi/6900800
**Failure Reason:** qa/standalone/osd/repro_long_log.sh failed
**Test that failed:** TEST_trim_max_entries_with_dups at line 172

**Comment by NitzanMordhai** [comment]
> **Teuthology Test Result** https://pulpito.ceph.com/yuriw-2022-06-27_15:15:17-rados-wip-yuri2-testing-2022-06-24-1331-distro-default-smithi/
> 
> **Related Failure:** @NitzanMordhai The following failed job appears to be related to this PR. Can you please take a look? **JobID:** https://pulpito.ceph.com/yuriw-2022-06-27_15:15:17-rados-wip-yuri2-testing-2022-06-24-1331-distro-default-smithi/6900800 **Failure Reason:** qa/standalone/osd/repro_long_log.sh failed **Test that failed:** TEST_trim_max_entries_with_dups at line 172

I see my mistake, correcting it

**Comment by NitzanMordhai** [comment]
jenkins test api

**Comment by NitzanMordhai** [comment]
jenkins test windows

**Comment by NitzanMordhai** [comment]
https://pulpito.ceph.com/yuriw-2022-07-09_14:48:58-rados-wip-yuri6-testing-2022-07-07-1429-distro-default-smithi/

Failures, unrelated:
6921591 - https://tracker.ceph.com/issues/52321
6921593 - https://tracker.ceph.com/issues/52321
6921594 - https://tracker.ceph.com/issues/55789
6921595 - https://tracker.ceph.com/issues/52321
6921596 - https://tracker.ceph.com/issues/52321
6921597 - https://tracker.ceph.com/issues/55789

Details:
6921591\6921593\6921595\6921596 - qa/tasks/rook times out: 'check osd count' reached maximum tries (90) after waiting for 900 seconds - Ceph - Rook
6921594\6921597 -  cls_rgw.index_suggest_complete assert failuer during test

**Pull request merged**